atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweeperService.java

```diff
@@ -27,35 +29,37 @@
  * Provides endpoints for sweeping a specific table.
  */
 @Path("/sweep")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
 public interface SweeperService {
-    /**
-     * Sweep a particular table from EMPTY startRow with default {@link SweepBatchConfig}.
-     */
-    @POST
-    @Path("sweep-table")
-    @Produces(MediaType.APPLICATION_JSON)
-    void sweepTable(@QueryParam("tablename") String tableName);
 
-    /**
-     * Sweep a particular table from specified startRow with default {@link SweepBatchConfig}.
-     */
-    @POST
-    @Path("sweep-table-from-row")
-    @Produces(MediaType.APPLICATION_JSON)
-    void sweepTableFromStartRow(
-            @QueryParam("tablename") String tableName,
-            @Nonnull @QueryParam("startRow") String startRow);
+    default SweepTableResponse sweepTable(String tableName) {
+        return sweepTable(tableName, Optional.empty(), Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty());
+    }
+
+    default SweepTableResponse sweepTableFrom(String tableName, String startRow) {
+        return sweepTable(tableName, Optional.of(startRow), Optional.empty(),
+                Optional.empty(), Optional.empty(), Optional.empty());
+    }
 
     /**
-     * Sweep a particular table from specified startRow with specified {@link SweepBatchConfig} parameters.
+     * Sweeps a particular table
+     *
+     * @param tableName the table to sweep
+     * @param startRow (Optional) the row to start from
+     * @param fullSweep (Optional; default true) whether to sweep the full table; if false just runs one batch
+     * @param maxCellTsPairsToExamine (Optional) see {@link SweepBatchConfig#maxCellTsPairsToExamine()}
+     * @param candidateBatchSize (Optional) see {@link SweepBatchConfig#candidateBatchSize()}
+     * @param deleteBatchSize (Optional) see {@link SweepBatchConfig#deleteBatchSize()}
      */
     @POST
-    @Path("sweep-table-from-row-with-batch")
-    @Produces(MediaType.APPLICATION_JSON)
-    void sweepTableFromStartRowWithBatchConfig(
+    @Path("sweep-table")
+    SweepTableResponse sweepTable(
             @QueryParam("tablename") String tableName,
-            @Nullable @QueryParam("startRow") String startRow,
-            @Nullable @QueryParam("maxCellTsPairsToExamine") Integer maxCellTsPairsToExamine,
-            @Nullable @QueryParam("candidateBatchSize") Integer candidateBatchSize,
-            @Nullable @QueryParam("deleteBatchSize") Integer deleteBatchSize);
+            @QueryParam("startRow") Optional<String> startRow,
+            @QueryParam("fullSweep") @DefaultValue("true") Optional<Boolean> fullSweep,
+            @QueryParam("maxCellTsPairsToExamine") Optional<Integer> maxCellTsPairsToExamine,
+            @QueryParam("candidateBatchSize") Optional<Integer> candidateBatchSize,
+            @QueryParam("deleteBatchSize") Optional<Integer> deleteBatchSize);
```

Comment:
Also, while we are here, can we marke these params etc. @Safe for logging purposes?